src/components/views/spaces/SpacePanel.tsx

```diff
@@ -204,58 +205,89 @@ const SpacePanel = () => {
+                                            tooltip={_t("All rooms")}
+                                            notificationState={RoomNotificationStateStore.instance.globalState}
+                                            isNarrow={isPanelCollapsed}
+                                        />
+                                        { invites.map(s => (
+                                            <SpaceItem
+                                                key={s.roomId}
+                                                space={s}
+                                                activeSpaces={activeSpaces}
+                                                isPanelCollapsed={isPanelCollapsed}
+                                                onExpand={() => setPanelCollapsed(false)}
+                                            />
+                                        )) }
+                                        { spaces.map((s, i) => (
+                                            <Draggable key={s.roomId} draggableId={s.roomId} index={i}>
+                                                {(provided, snapshot) => (
+                                                    <SpaceItem
+                                                        {...provided.draggableProps}
+                                                        {...provided.dragHandleProps}
+                                                        key={s.roomId}
+                                                        innerRef={provided.innerRef}
+                                                        className={snapshot.isDragging
+                                                            ? "mx_SpaceItem_dragging"
+                                                            : undefined}
+                                                        space={s}
+                                                        activeSpaces={activeSpaces}
+                                                        isPanelCollapsed={isPanelCollapsed}
+                                                        onExpand={() => setPanelCollapsed(false)}
+                                                    />
+                                                )}
+                                            </Draggable>
+                                        )) }
+                                        { provided.placeholder }
+                                    </div>
+                                    <SpaceButton
+                                        className={newClasses}
+                                        tooltip={menuDisplayed ? _t("Cancel") : _t("Create a space")}
+                                        onClick={menuDisplayed ? closeMenu : () => {
+                                            if (!isPanelCollapsed) setPanelCollapsed(true);
+                                            openMenu();
+                                        }}
```

Comment:
This DOM tree is massive... can we break it out stuff like listeners and potentially whole elements to their own variables please?

Reply:
Not much whilst decreasing the visible depth given the number of render-props